app/controllers/base/model_controller.rb

```diff
@@ -7,6 +7,32 @@ class ModelController < BaseController
 
     attr_reader :object_renderer, :collection_renderer
 
+    def initialize(config, logger, env, params, body, sinatra=nil, dependencies={})
+      validate_parameters(params)
+      super(config, logger, env, params, body, sinatra, dependencies)
+    end
+
+    def validate_parameters(params)
+      valid_params = {}
+      [
+        ['order-by',               String],
+
+      ].each do |key, klass|
+        val = params[key]
+        valid_params[key.underscore.to_sym] = Object.send(klass.name, val) if val
+      end
+
+      if valid_params[:order_by]
+        valid_params[:order_by].split(',').each do |col|
+          unless self.class.query_parameters.to_a.include?(col)
+            raise VCAP::Errors::ApiError.new_from_details(
+                'BadQueryParameter',
+                "invalid request parameter '#{col}' in order_by")
+          end
+        end
+      end
+    end
+
```

Comment:
This validation should live outside the controller. `OrderApplicator` looks like the correct home. Look at [query.rb](https://github.com/cloudfoundry/cloud_controller_ng/blob/26b6cdc812800f62c8166444ca4feff79cc38316/lib/vcap/rest_api/query.rb#L57) for an example of injecting the controller's `query_parameters` into another class to do the validation.

This will also reduce the complexity of the validation (`valid_params` shouldn't be necessary to build), since you will only need to worry about the `order-by` parameter.